### PR TITLE
Generate windows training package

### PR DESCRIPTION
**Pull request by baijumeswani**
This pull request adds an option to build a windows package for `onnxruntime-training` for x64.

### CODE REVIEW

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml [hunk 1]
```diff
@@ -417,22 +417,25 @@ stages:
...
-            PythonVersion: '3.9'
-            MsbuildPlatform: Win32
-            buildArch: x86
-          Python310_x86:
-            PythonVersion: '3.10'
-            MsbuildPlatform: Win32
-            buildArch: x86
+          # Training build cannot support Win32 for now because one or more of its python
+          # dependencies does not support Win32. So, don't build a training package for Win32
+          ${{ if not(contains(parameters.build_py_parameters, '--enable_training')) }}:
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:422]: I would suggest adding it to regular builds. It also help us find problems earlier.

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml [hunk 2]
```diff
@@ -417,22 +417,25 @@ stages:
...
+            Python37_x86:
+              PythonVersion: '3.7'
+              MsbuildPlatform: Win32
```
Comment (snnn) [inline tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:425]: So you need a 32-bit build instead of 64-bit? Why? Who need this?

Comment (baijumeswani) [reply tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:425]: We need only x64. So, we are checking if it is a training build, and skipping build for x86 in it is a training build.

File: tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml (hunk 1 above)
Comment (baijumeswani) [reply tools/ci_build/github/azure-pipelines/templates/py-packaging-stage.yml:422]: You mean have regularly scheduled builds for `onnxruntime-training` for `Windows`? I can add a schedule for that. We don't want to publicly expose that yet, but it is ok to build the package and consume internally.

**Review by snnn — APPROVED** [review]

**Pull request merged**